relayer/src/lib.rs

```diff
@@ -231,20 +237,22 @@ pub async fn submit_empty_block_loop(
     gas_limit: u64,
     empty_block_interval: Duration,
 ) -> Result<(), Error> {
+    let web_state = WebState::new(contract, nonce_count_rule, gas_limit);
```

Comment:
This will create a new `Mutex` which doesn't synchronize with the one used in the request handler. I think this function should take a (_the_) `WebState` as an argument.